thrift/compiler/lib/schematizer.cc

```diff
@@ -761,7 +765,11 @@ std::string_view schematizer::program_checksum(const t_program& program) {
   // @lint-ignore CLANGTIDY facebook-hte-CArray
   unsigned char hash[MD5_DIGEST_LENGTH];
   auto val = sm_.get_file(program.path())->text;
+#if OPENSSL_VERSION_NUMBER >= 0x030000000  // v3.0.0
+  EVP_Digest(val.data(), val.size(), hash, nullptr, EVP_md5(), nullptr);
+#else
   MD5(reinterpret_cast<const unsigned char*>(val.data()), val.size(), hash);
```

Comment:
Why is `MD5` deprecated? It would seem odd to deprecate direct functions like this in favor of indirect layers around them.

Reply:
https://github.com/openssl/openssl/issues/12260#issuecomment-649404727

Reply:
I don't think we necessarily need to make the code as ugly as it is. The EVP API has been available since openssl-1.1.0 (August 2016), so I'd think it's fair to assume its existence and drop the preprocessor checks.